<launch>
  
  <arg name="id" default="0" />
  <arg name="robot" default="robot_$(arg id)" />
  
  <arg name="random" default="false" />
  <arg name="x_min" default="-9.0" />
  <arg name="x_max" default="9.0" />
  <arg name="y_min" default="-6.0" />
  <arg name="y_max" default="6.0" />
  
  <group ns="$(arg robot)">
    <param name="tf_prefix" value="$(arg robot)" />

	  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    	  <rosparam file="$(find soccer_stage)/config/costmap_common_params.yaml" command="load" ns="global_costmap" />
    	 	<rosparam file="$(find soccer_stage)/config/costmap_common_params.yaml" command="load" ns="local_costmap" />
    		<rosparam file="$(find soccer_stage)/config/local_costmap_params.yaml" command="load" />
    		<rosparam file="$(find soccer_stage)/config/global_costmap_params.yaml" command="load" />
    		<rosparam file="$(find soccer_stage)/config/teb_local_planner_params.yaml" command="load" />

		    <param name="base_global_planner" value="global_planner/GlobalPlanner" />
		    <param name="planner_frequency" value="1.0" />
		    <param name="planner_patience" value="5.0" />

		    <param name="base_local_planner" value="teb_local_planner/TebLocalPlannerROS" />
		    <param name="controller_frequency" value="10.0" />
		    <param name="controller_patience" value="20.0" />  
	  </node>    
      
    <node pkg="fake_localization" type="fake_localization" name="fake_localization" respawn="false" output="screen">
        <param name="odom_frame_id" value="$(arg robot)/odom" />
        <param name="base_frame_id" value="$(arg robot)/base_link" />
    </node>
    
    <node pkg="tf" type="static_transform_publisher" name="map_broadcaster" args="0 0 0 0 0 0 1 map $(arg robot)/map 100" />
    
    <node pkg="soccer_stage" type="publish_ground_truth_obstacles.py" name="obstacles_publisher" output="screen">
        <param name="id" value="$(arg id)" />
    </node>
    
    <group if="$(arg random)">
      <!-- Move randomly -->
      <node name="move_random" pkg="soccer_stage" type="move_random.py" output="screen">
        <param name="x_min" value="$(arg x_min)" />
        <param name="x_max" value="$(arg x_max)" />
        <param name="y_min" value="$(arg y_min)" />
        <param name="y_max" value="$(arg y_max)" />
      </node>    
    </group>
    
  </group>

</launch>
